### ISSUE TITLE
[FlexAttention] Error using create_block_mask with mask head number greater than 1

### ISSUE THREAD

**Issue by ChenlongDeng** [pytorch/pytorch#148527]
### 🐛 Describe the bug

I have encountered an error when using the flex_attention function in combination with a block mask.  Specifically, the error occurs when the mask `created by create_block_mask` is configured with a head number greater than 1. If the mask head number is set to 1, the code runs without any issues.

To reproduce this problem, I have provided the code snippet below.  This code is self-contained and should directly reproduce the error I am experiencing.

```python
from torch.nn.attention.flex_attention import create_block_mask, flex_attention
import torch

flex_attention = torch.compile(flex_attention, dynamic=False)

q = torch.randn(1, 32, 8, 128, dtype=torch.bfloat16, device="cuda:0")
k = torch.randn(1, 32, 8, 128, dtype=torch.bfloat16, device="cuda:0")
v = torch.randn(1, 32, 8, 128, dtype=torch.bfloat16, device="cuda:0")

def easy_head_attention_mod(head_num):
    head_type = torch.tensor([False if i % head_num == 0 else True for i in range(head_num)], dtype=torch.bool, device=q.device)

    def mask_mod(b, h, q_idx, kv_idx):
        bi_mask = True & head_type[h]
        causal_mask = q_idx >= kv_idx

        return bi_mask & causal_mask

    return mask_mod


mask_mod = easy_head_attention_mod(32) # Error occurs when head_num is greater than 1, e.g., 32
# If head_num is set to 1 (e.g., mask_mod = easy_head_attention_mod(1)), the code runs without error

mask = create_block_mask(mask_mod, 1, 32, 8, 8, device=q.device, _compile=True)

# Use `enable_gqa=True` with corresponding inputs here would bring more bugs
attn_output = flex_attention(q, k, v, block_mask=mask)

print(attn_output.shape)
```

Upon running the code, the following Assertion Error is raised:

```shell
/tmp/torchinductor_root/hp/chpkqukqm77fa3dop4cafoobwxnw5allj4hh2ti37awkdk6gh34c.py:118: unknown: block: [0,3,0], thread: [62,0,0] Assertion `` failed.
/tmp/torchinductor_root/hp/chpkqukqm77fa3dop4cafoobwxnw5allj4hh2ti37awkdk6gh34c.py:118: unknown: block: [0,3,0], thread: [63,0,0] Assertion `` failed.
...
```

Thank you for your time and attention to this issue. I hope this report is helpful in identifying and resolving the problem.

### Versions

torch==2.7.0.dev20250302+cu118

**Comment by drisspg** [comment]
This is a problem w/ our FlexDecoding kernel which is being triggered from the small seq-len as a work around can you  run w/ 

`attn_output = flex_attention(q, k, v, block_mask=mask, kernel_options={'FORCE_USE_FLEX_ATTENTION': True})`

cc @BoyuanFeng
